unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BuildModeHUD/Scripts/BuildModeHUDController.cs

```diff
@@ -166,9 +176,68 @@ private void ConfigureCatalogItemDropController()
         catalogItemDropController.OnCatalogItemDropped += CatalogItemSelected;
     }
 
-    public void PublishStart() { view.PublishStart(); }
+    private void ConfigureBuildModeConfirmationModalController()
+    {
+        controllers.buildModeConfirmationModalController.OnCancelExit += CancelPublishModal;
+        controllers.buildModeConfirmationModalController.OnConfirmExit += ConfirmPublishModal;
+    }
+
+    public void PublishStart()
+    {
+        controllers.buildModeConfirmationModalController.Configure(
+            PUBLISH_MODAL_TITLE,
+            PUBLISH_MODAL_SUBTITLE,
+            PUBLISH_MODAL_CANCEL_BUTTON,
+            PUBLISH_MODAL_CONFIRM_BUTTON);
+        controllers.buildModeConfirmationModalController.SetActive(true, BuildModeModalType.PUBLISH);
+    }
+
```

Comment:
Maybe you're right. Anyway I prefer to wait for the future needs before doing any change. For now I like to follow this approach because it is the same that the other controllers are following.

We will be attentive to future needs. Thanks Adri!